ql/src/java/org/apache/hadoop/hive/ql/parse/TaskCompiler.java

```diff
@@ -204,6 +219,16 @@ public void compile(final ParseContext pCtx,
           fetch.setIsUsingThriftJDBCBinarySerDe(false);
       }
 
+      Collection<Operator<? extends OperatorDesc>> tableScanOps =
+          Lists.<Operator<?>>newArrayList(pCtx.getTopOps().values());
+      Set<FileSinkOperator> fsOps = OperatorUtils.findOperators(tableScanOps, FileSinkOperator.class);
+      if(fsOps != null && fsOps.size() == 1) {
+        FileSinkOperator op = fsOps.iterator().next();
+        Set<Path> filesToFetch =  new HashSet<>();
+        op.getConf().setFilesToFetch(filesToFetch);
```

Comment:
I am not sure how this works. filesToFetch field is populated in tasks. Having this same reference won't help, because plan (and FilesinkDesc) is sent to cluster from HS2, but any changes in plan won't reflect here after job is finished on cluster.

Reply:
filesToFetch is populated on client(HS2) side not task side (during job close op) that is why having reference works.

Reply:
Right. Apologies for the confusion.